util/src/main/java/io/grpc/util/SpiffeIdParser.java

```diff
@@ -0,0 +1,10 @@
+package io.grpc.util;
+
+public interface SpiffeIdParser {
```

Comment:
Could you elaborate on why you chose to use an interface here? Offline you mentioned that this is so users can substitute their own version, but

- I don't know of a use case where we would want to allow a user to do so, because it seems like an easy way to let a user shoot themselves in the foot.
- I don't think we will add plumbing that would let the user inject their own parser.

Reply:
So the scenario I had in mind was - 

- provider setter for such parser (Interface) in gRPC TrustManager
- users can implement their own and set the one they want for gRPC TrustManager

However since Eric mentioned it's not a public API I'll drop this flexibility. Users can still implement their own X509ExtendedKey/TrustManager and do what's needed their (so there are still many ways to shoot themselves in the feet anyway)